source/common/tls/context_impl.cc

```diff
@@ -351,6 +351,17 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
       SSL_CTX_set_keylog_callback(ctx, keylogCallback);
     }
   }
+
+  // Compliance policy must be applied last to have a defined behavior.
+  switch (config.compliancePolicy()) {
+  case Envoy::Ssl::CompliancePolicy::FIPS_202205:
```

Comment:
I wonder if we should reject the config, or log a warning, if this is configured and FIPS_mode() isn't true?

Reply:
BoringSSL doesn't require FIPS_mode apparently, so we just conform to their API.

Reply:
That's true, but it seems like it's probably an unintended configuration; just trying to help people not do nonsensical things on accident.

Reply:
Added a warning. Speaking of which, we also should use FIPS_mode instead of BORINGSSL_FIPS pragmas in the rest of this file.